cluster-autoscaler/core/utils/utils.go

```diff
@@ -139,6 +140,41 @@ func sanitizeTemplateNode(node *apiv1.Node, nodeGroup string, ignoredTaints tain
 	return newNode, nil
 }
 
+// PodSpecSemanticallyEqual returns if two pod specs are similar after dropping
+// the fields we don't care about
+func PodSpecSemanticallyEqual(p1 apiv1.PodSpec, p2 apiv1.PodSpec) bool {
+	p1Spec := sanitizedProjectedVolumesAndMounts(p1)
+	p2Spec := sanitizedProjectedVolumesAndMounts(p2)
+	return apiequality.Semantic.DeepEqual(p1Spec, p2Spec)
+}
+
+// sanitizedProjectedVolumesAndMounts returns a pod spec with projected volumes
+// and their mounts removed
+// Due to the generated suffixes, a strict DeepEquals check will fail and generate
+// an equivalence group per pod which is undesirable.
+// Projected volumes do not impact scheduling so we should ignore them
+func sanitizedProjectedVolumesAndMounts(podSpec apiv1.PodSpec) apiv1.PodSpec {
+	var projectedVolumeNames []string
+	for i, v := range podSpec.Volumes {
+		if v.Projected != nil {
+			projectedVolumeNames = append(projectedVolumeNames, v.Name)
+			podSpec.Volumes[i] = apiv1.Volume{}
+		}
+	}
+
+	for _, pv := range projectedVolumeNames {
+		for c := range podSpec.Containers {
+			for v, mount := range podSpec.Containers[c].VolumeMounts {
+				if mount.Name == pv {
```

Comment:
couldn't think of a better way to go from volumes -> volumeMounts apart from the volume name.